### ISSUE TITLE
not paging

### ISSUE THREAD

**Issue by doubletaketech** [percolatestudio/paginated-subscription#2]
I'm using this with Meteor 1.0.3.  When scroll, does not paginate. I only see first 10 and nothing changes when scrolling.

What am I going wrong?

Thanks,
Don

```
Meteor.publish('itemsInfinite',function(limit){
        return Items.find({},{limit:limit});
    })

var handle = Meteor.subscribeWithPagination('itemsInfinite', 10);

Template.gridItems.helpers({
    items:function(){
        if(Session.get('searchTags')) {
            var tagArray = Session.get('searchTags');
            return Items.find({ tags: { $all: Session.get('searchTags') } },{limit:10})
        }
        else {
            return Items.find({},{limit:10})

        }

    }
});
```

**Comment by doubletaketech** [comment]
How to use the itemsInfinite subscribeWithPagination in my gridItems template helper?
adding limit:10 in my helper doesn't pick up the subscription, it just limits to 10 only.

Thanks

**Comment by guysmoilov** [comment]
Hi,
just from reading the readme, it seems to me that you don't need to supply the limit in every Items.find. In fact, that will definitely always limit you to just 10 Items.
You should just use the subscription as you did above and use find without limit, the paginated publish will limit it for you. If you want more Items, just call handle.loadNextPage() and they should appear on the screen automagically (because behind the scenes the publish function will just send you more Items).

Again, I never actually tried this package yet, so I'm sorry if I'm talking nonsense.

**Comment by v3rron** [comment]
@guysmoilov yeah, whatever you say is absolutely makes sense, but unfortunately it doesn't work this way. Just tried this package with Meteor 1.1.0.3 and it publishes all the data to the client even with the limit set :(
PS: It would be nice to have a real life example working with Iron-Router, waitOn and InfiniteScroll.

**Comment by tmeasday** [comment]
You'd want to pass `{limit: handle.loaded}` there I'd say.

**Comment by v3rron** [comment]
In case someone is interested to have the subscription in IR(Iron Router) router file with waitOn hook in order to get loadingTemplate working, here's how I managed to do it with IR controller:

```
Router.route('posts', {
  controller: RouteController.extend({
    waitOn: function () {
      return [
        Meteor.subscribeWithPagination('allPosts', 20),
        Meteor.subscribeWithPagination('allComments', 20)
      ];
    },
  }),
});
```

Note: `allPosts` and `allComments` are publish names.
And then in my template:

```
Template.posts.onCreated(function () {
  var tmpl = this;
  tmpl.postHandler = Iron.controller().waitOn()[0];
  tmpl.commentsHandler = Iron.controller().waitOn()[1];
});

Template.posts.helpers({
  all_posts: function(){
    var tmpl = Template.instance();
    return Posts.find({}, {sort: {updatedAt: -1}, limit: tmpl.postHandler.loaded()});
  },
  all_comments: function(){
    var tmpl = Template.instance();
    return Comments.find({}, {sort: {createdAt: 1}, limit: tmpl.commentsHandler.loaded()});
  }
});
```

PS: @tmeasday, just curious, why do we still need to provide limit in helpers if we already did that on subsription? Shouldn't it be already limited on subscription level?
Thanks in advance.
